### DISCUSSION TITLE
How to turn off dependabot PRs but still get alerts?

### DISCUSSION THREAD

**Discussion by nello** [community/community#20354] [Code Security] [answered]
The control of this feature is incredibly confusing, at least to me. Reading the documentation, I would expect that alerting could be turned on quite simply but that I would then have to create a YAML file to configure automatic patching PRs. But somehow the converse is true. 

I have **never** turned on anything to do with PRs yet they get submitted constantly for one of my repos, even though I have toggled them off in the UI. How is this _meant_ to work?

<img width="804" alt="image" src="https://user-images.githubusercontent.com/966106/178375158-0de7bea1-3319-49dd-894d-aaca2b3e6d89.png">

**Comment by shmulik-klein** [comment, marked as answer]
Try to use the main `settings` page under you profile - with `Disable All` (and make sure the checkbox is off)

![image](https://user-images.githubusercontent.com/18899126/178715791-ed69f744-faec-4759-b219-10f56f25b814.png)

**Reply by nello** [reply]
What I would _like_ to do is get `dependabot` alerts for my dependencies but _not_ have it automatically open pull requests. Is there some magical combination of incantations that does this? Forgive me if I'm not understanding, but your answer just seems like a way to turn it off.

**Reply by shmulik-klein** [reply]
Yep, I just meant that you'll disable _Dependabot security updates_, this option controls whether it should open PRs, the other option will keep you updated.

I hope it helps 🤞

**Reply by jhutchings1** [reply]
@shmulik-klein is correct. The `Dependabot security updates` setting configures whether or not Dependabot opens pull requests. The `Dependabot alerts` setting configures whether you receive alerts. Hope this helps!

**Reply by jhutchings1** [reply]
I should add that _if_ you have a dependabot.yml file checked into your repository, then that tells Dependabot to send pull requests as well. Delete that file and you'll clear out that configuration.

**Reply by nello** [reply]
Thanks for the extra information. I've deleted all dependabot yamls for our repos now. They appear to have been added by people who thought adding them was best practice without realising the consequences.

I really do appreciate your help but this would have been much easier to resolve with if a) Github's controls were labelled properly (seriously "Dependabot security updates" could mean just about anything!), b) the controls were scoped correctly so that repo controls were not overridden by global ones, and c) the mere presence of a file didn't override all other settings.

**Reply by jhutchings1** [reply]
Appreciate the feedback. I've relayed that feedback to the team, because I agree that the UI is super confusing when a config file is checked in. We'll look at whether there are clearer messages we can provide here. Thanks!

**Comment by AndreaGriffiths11** [comment]
Adding that, you can always customize how dependencies are maintained: check out https://www.youtube.com/watch?v=Qcq0WfXz-aM&list=PL0lo9MOBetEG2cjOxhvHNVS4NUYiZPm6z
and [https:github.co/likeapro](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file)

**Comment by nello** [comment]
Thank for your responses, but honestly this is way too hard. I don't want to have to create and maintain a yaml for Github to not make unwarranted PRs against my codebases. I don't want to go searching outside my repo for ways to make the dependency-checking configuration in the repo work in an expected way. I've turned everything to do with dependency checking from Github off for all my repos wherever I can find it and will rely on outside tooling from now on.
